stats/stats/src/charts/types/new_txns.rs

```diff
@@ -0,0 +1,88 @@
+use chrono::NaiveDate;
+use sea_orm::FromQueryResult;
+
+use crate::{data_source::kinds::data_manipulation::map::MapFunction, ChartError};
+
+use super::{TimespanTrait, TimespanValue, ZeroTimespanValue};
+
+#[derive(FromQueryResult, Clone, Debug, PartialEq)]
+pub struct NewTxnsCombinedPoint {
+    pub date: NaiveDate,
+    pub all_transactions: String,
+    pub op_stack_operational_transactions: String,
+}
+
+impl TimespanTrait for NewTxnsCombinedPoint {
+    type Timespan = NaiveDate;
+
+    fn timespan(&self) -> &Self::Timespan {
+        &self.date
+    }
+
+    fn timespan_mut(&mut self) -> &mut Self::Timespan {
+        &mut self.date
+    }
+}
+
+impl ZeroTimespanValue<NaiveDate> for NewTxnsCombinedPoint {
+    fn with_zero_value(timespan: Self::Timespan) -> Self {
+        Self {
+            date: timespan,
+            all_transactions: "0".to_string(),
+            op_stack_operational_transactions: "0".to_string(),
+        }
+    }
+}
+
+pub struct ExtractAllTxns;
+
+impl MapFunction<Vec<NewTxnsCombinedPoint>> for ExtractAllTxns {
+    type Output = Vec<TimespanValue<NaiveDate, String>>;
+
+    fn function(inner_data: Vec<NewTxnsCombinedPoint>) -> Result<Self::Output, ChartError> {
+        Ok(inner_data
+            .into_iter()
+            .map(|p| TimespanValue {
+                timespan: p.date,
+                value: p.all_transactions,
+            })
+            .collect())
+    }
+}
+
+impl MapFunction<NewTxnsCombinedPoint> for ExtractAllTxns {
+    type Output = TimespanValue<NaiveDate, String>;
+
+    fn function(inner_data: NewTxnsCombinedPoint) -> Result<Self::Output, ChartError> {
+        Ok(TimespanValue {
+            timespan: inner_data.date,
+            value: inner_data.all_transactions,
+        })
+    }
+}
+
+pub struct ExtractOpStackTxns;
```

Comment:
imho this struct should be defined in `op_stack_operational_transactions.rs` not here

Reply:
u defined it here and use it `op_stack_operational_transactions.rs`

Reply:
<img width="370" alt="image" src="https://github.com/user-attachments/assets/c8757c25-e6f8-453f-b0ea-dc6978367593" />
It's also used in other places though?
I thought it's weird to define a struct in one chart's file and then use in another (just like we agreed to make subfolders for such cached charts)

mod.rs is another candidate but it's also kinda weird idk